sqlglot/parser.py

```diff
@@ -572,93 +539,145 @@ def parse_option(option, token, option_lambda):
-
-            if schema and not isinstance(value, exp.Schema):
-                columns = {v.text("this").upper() for v in value.expressions}
-                partitions = [
-                    expression
-                    for expression in schema.expressions
-                    if expression.this.text("this").upper() in columns
-                ]
-                schema.set(
-                    "expressions",
-                    [e for e in schema.expressions if e not in partitions],
-                )
-                value = self.expression(exp.Schema, expressions=partitions)
-        else:
-            value = self._parse_column()
-
-        return self.expression(
-            exp.Property,
-            this=exp.Literal.string(key),
-            value=value,
-        )
-
-    def _parse_properties(self, schema):
-        properties = []
+        if self._match(TokenType.USING):
+            return self.expression(
+                exp.TableFormatProperty,
+                this=exp.Literal.string(c.TABLE_FORMAT),
+                value=exp.Literal.string(self._parse_var().name),
+            )
+        if self._match(TokenType.PARTITIONED_BY):
+            return self.expression(
+                exp.PartitionedByProperty,
+                this=exp.Literal.string(c.PARTITIONED_BY),
+                value=self._parse_schema(),
+            )
+        if self._match(TokenType.STORED):
+            self._match(TokenType.ALIAS)
+            self._match(TokenType.EQ)
+            return self.expression(
+                exp.FileFormatProperty,
+                this=exp.Literal.string(c.FILE_FORMAT),
+                value=exp.Literal.string(self._parse_var().text("this")),
+            )
+        if self._match(TokenType.LOCATION):
+            return self.expression(
+                exp.LocationProperty,
+                this=exp.Literal.string(c.LOCATION),
+                value=self._parse_string(),
+            )
+        if self._match(TokenType.FORMAT):
+            self._match(TokenType.EQ)
+            return self.expression(
+                exp.FileFormatProperty,
+                this=exp.Literal.string(c.FILE_FORMAT),
+                value=self._parse_string() or self._parse_var(),
+            )
+        if self._match(TokenType.ENGINE):
+            self._match(TokenType.EQ)
+            return self.expression(
+                exp.EngineProperty,
+                this=exp.Literal.string(c.ENGINE),
+                value=self._parse_var() or self._parse_string(),
+            )
+        if self._match(TokenType.AUTO_INCREMENT):
+            self._match(TokenType.EQ)
+            return self.expression(
+                exp.AutoIncrementProperty,
+                this=exp.Literal.string(c.AUTO_INCREMENT),
+                value=self._parse_var() or self._parse_number(),
+            )
+        if self._match(TokenType.CHARACTER_SET) or self._match_sequence(
```

Comment:
this is the only one that won't map in the dictionary, but that's ok, just have that one after the main dictionary matcher